File: 2024/09 September/Reverse Words/main.cpp
// Link: https://www.geeksforgeeks.org/problems/reverse-words-in-a-given-string5459/1

#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    string reverseWords(string str)
    {
        vector<string> temp;

        string t = "";

        for (int i = 0; i < str.length(); i++)
        {
            if (str[i] == '.')
            {
                temp.push_back(t);
                t = "";
            }
            else
            {
                t += str[i];
            }
            if (i == str.length() - 1)
            {
                temp.push_back(t);
            }
        }

        string ans = "";

        for (int i = temp.size() - 1; i >= 0; i--)
        {
            ans += temp[i];
            if (i != 0)
            {
                ans += '.';
            }
        }

        return ans;
    }
};

int main()
{
    Solution obj;
    string str = "i.like.this.program.very.much";
    cout << obj.reverseWords(str) << endl;

    str = "pqr.mno";
    cout << obj.reverseWords(str) << endl;
    return 0;
}

File: 2024/09 September/Reverse Words/main.py
# Link: https://www.geeksforgeeks.org/problems/reverse-words-in-a-given-string5459/1


class Solution:
    def reverseWords(self, str):
        temp = []
        t = ""

        for i in range(len(str)):
            if str[i] == ".":
                temp.append(t)
                t = ""
            else:
                t += str[i]
            if i == len(str) - 1:
                temp.append(t)

        ans = ""

        for i in range(len(temp) - 1, -1, -1):
            ans += temp[i]
            if i != 0:
                ans += "."

        return ans


if __name__ == "__main__":
    obj = Solution()
    str1 = "i.like.this.program.very.much"
    print(obj.reverseWords(str1))

    str2 = "pqr.mno"
    print(obj.reverseWords(str2))
